### DISCUSSION TITLE
cannot pass variable to resolveComponent().

### DISCUSSION THREAD

**Discussion by masayu36** [nuxt/nuxt#17661] [Questions]
In the latest version it's not possible to use "component :is" like before.

Before, it was possible to set component dynamically like this:
<template>
 <div>
  <component :is="anyComponent" />
 </div>
</template>

<script setup lang="ts">
 const anyComponent = 'test1' 
 // or
 // const str = 'test1'
 // const anyComponent = str
</script>

But, the updated document says you have to use "resolveComponent()" when you use :is.
https://v3.nuxtjs.org/guide/directory-structure/components#dynamic-components

<template>
 <div>
  <component :is="anyComponent" />
 </div>
</template>

<script setup lang="ts">
 const anyComponent = resolveComponent('test')
</script>

This works well. 
But when you try to pass variable to resolveComponent(), an error like this occurs: 'Fail to resolve component...If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.'
<script setup lang="ts">
 const str = 'test1'
 const anyComponent = resolveComponent(str)  // fail to resolve component
</script>

Is there any solution? Or a bug?

Thanks in advance.

**Comment by riso348** [comment]
Hi there,
this is something that i also face in new nuxt3, currently, as a workaround i found, that you can use component based on variable names, when you register it globally..

```
import { defineNuxtConfig } from 'nuxt'

  export default defineNuxtConfig({
    components: {
     global: true,
     dirs: ['~/components']
    },
  })
```

after this, you will be able fetch component name throu api, and show it:

```
<template>
  <div>
    <component :is="realComponent"></component>
  </div>
</template>

<script setup lang="ts">
import {load} from "~/composables/page";
import {resolveComponent} from "@vue/runtime-core";


const {pageData, sections, dynComponent} = await load();
const realComponent = resolveComponent(dynComponent)

</script>

```

while in my case, the load function load data from api using useFetch
